client/components/Profile.jsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Loader from "../components/Loader.jsx"
+
+const profileSubs = new SubsManager();
+
+const Profile = React.createClass({
+  mixins: [ReactMeteorData],
+
+  getInitialState: function() {
+    return {
+      nicknameChanged: false
+    };
+   },
+
+  getMeteorData () {
+    const usersHandle = profileSubs.subscribe("users");
+    let user = {};
+
+    if (usersHandle.ready()) {
+      user = Meteor.users.findOne(this.props.userProfile);
+    }
+    return {
+      usersHandle,
+      user
+    }
+  },
+
+  handleSubmit(e) {
+    e.preventDefault();
+    const nickname = ReactDOM.findDOMNode(this.refs.nickname).value.trim();
+    if (nickname.length === 0) return;
+    Meteor.call("user.changeNickname", nickname, (error, result) => {
+      if (!error) {
+        this.setState({nicknameChanged: true})
+      }
+    });
+  },
+
+  componentDidMount() {
+    if (this.data.usersHandle.ready() && this.props.userProfile == Meteor.userId()) {
```

Comment:
Use ===.
this.data.usersHandle.ready() is a reactive source, meaning it might not be true when you call it. ComponentDidMount is called only once, so if the handle is not ready, the form will never initialize. You can wrap this in a Tracker.autorun and keep a this.formInitialized boolean variable to prevent multiple initializations.